lib/obsolete.gd

```diff
@@ -479,6 +479,30 @@ DeclareObsoleteSynonym( "MutableIdentityMat", "IdentityMat", "4.8" );
 ##
 DeclareObsoleteSynonym( "MutableNullMat", "NullMat", "4.8" );
 
+#############################################################################
+##
+#F  IsSemilatticeAsSemigroup( <S> ) is the semigroup <S> a semilattice
+##
+##  <#GAPDoc Label="IsSemilatticeAsSemigroup">
+##  <ManSection>
+##  <Prop Name="IsSemilatticeAsSemigroup" Arg='S'/>
+##
+##  <Description>
+##    <C>IsSemilatticeAsSemigroup</C> returns <K>true</K> if the semigroup
+##    <A>S</A> is a semilattice and <K>false</K> if it is not. <P/>
+##
+##    A semigroup is a <E>semilattice</E> if it is commutative and every
+##    element is an idempotent. The idempotents of an inverse semigroup form a
+##    semilattice.
+##
+##    This is identical to <Ref Prop="IsSemilattice" BookName = "Semigroups"/> #
+##    and is present in &GAP;&nbsp;4.8 #  only for the sake of compatibility with
+##    beta-releases.  #  It should <E>not</E> be used in new code.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareObsoleteSynonym( "IsSemilatticeAsSemigroup", "IsSemilattice", "4.8" );
```

Comment:
So this is still present, and in fact it breaks the `semigroups` package: With this branch, I get this:
```
gap> LoadPackage("semigroups");
Error, <oper> must be an operation in
  imp[2] := FLAGS_FILTER( from ); at /Users/mhorn/Projekte/GAP/gap.github/lib/filter.g:131 called from
InstallTrueMethodNewFilter( tofilt, from
 ); at /Users/mhorn/Projekte/GAP/gap.github/lib/filter.g:182 called from
<function "InstallTrueMethod">( <arguments> )
 called from read-eval loop at line
45 of /Users/mhorn/Projekte/GAP/repos/pkg/semigroups.remotehg/gap/properties.gd
Error, was not in any namespace called from
LEAVE_NAMESPACE(  ); at /Users/mhorn/Projekte/GAP/gap.github/lib/package.gi:1217 called from
<function "ReadPackage">( <arguments> )
 called from read-eval loop at line 60 of /Users/mhorn/Projekte/GAP/repos/pkg/semigroups.remotehg/init.g
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk> IsSemilatticeAsSemigroup;
function( arg ) ... end
```
So I think this should indeed be changed to `DeclareSynonymAttr` as I suggested.